apps/next/src/pages/Settings/components/NetworkManagement/components/NetworkDetailsFlow.tsx

```diff
@@ -1,17 +1,120 @@
+import { useEditNetwork } from '../hooks/useEditNetwork';
+
+type NetworkDetailsParams = {
+  networkId: string;
+};
 
 export const NetworkDetailsFlow = () => {
-  return (
-    <Stack
-      height="100cqh"
-      width={1}
-      bgcolor="background.backdrop"
-      overflow="hidden"
-      sx={{ position: 'relative' }}
-    >
-      <PageTopBar showBack={true} />
-      <div>NetworkDetailsFlow</div>;
-    </Stack>
-  );
+  const history = useHistory();
+  const { t } = useTranslation();
+  const { networkId } = useParams<NetworkDetailsParams>();
+
+  const [tab, setTab] = useState<EditNetworkFormTab>('details');
+
+  const {
+    network,
+    isValid,
+    isEditing,
+    setIsEditing,
+    isCustom,
+    removeCustomNetwork,
+    fieldInfo,
+    setNetwork,
+    reset,
+    resetRpcUrl,
+    submit,
+  } = useEditNetwork({
+    networkId: Number(networkId),
+    rpcUrlResetButtonAction: () => {
+      setTab('rpc-url-reset');
+    },
+  });
+
+  const resetRpcUrlHandler = async () => {
+    try {
+      await resetRpcUrl();
+      toast.success(t('RPC URL reset successfully'));
+      setTab('details');
+    } catch (error) {
+      console.error(error);
+      toast.error(t('Failed to reset RPC URL'));
+    }
+  };
+
+  const saveHandler = async () => {
+    try {
+      await submit();
+      toast.success(t('Network updated'));
+      setTab('details');
+    } catch (error) {
+      console.error(error);
+      toast.error(t('Failed to save network'));
+    }
+  };
+
+  const deleteHandler = async () => {
+    try {
+      await removeCustomNetwork(Number(networkId));
+      toast.success(t('Network deleted'));
+      history.goBack();
+    } catch (error) {
+      console.error(error);
+      toast.error(t('Failed to delete network'));
+      setTab('details');
+    }
+  };
+
+  return !network ? (
+    <NetworkNotFound />
+  ) : tab === 'details' ? (
+    <NetworkDetails
+      network={network}
+      setNetwork={setNetwork}
+      setTab={setTab}
+      onSubmit={() => setTab('save')}
+      onCancel={reset}
+      isValid={isValid}
+      fieldInfo={fieldInfo}
+      canResetRpcUrl={!isCustom}
+      autoFocus={false}
+      isEditing={isEditing}
+      setIsEditing={setIsEditing}
+      onDelete={() => setTab('delete')}
+      isCustom={isCustom}
+      pageType={'edit'}
+    />
+  ) : tab === 'rpc-headers' ? (
+    <CustomRpcHeadersManager
+      network={network}
+      setTab={setTab}
+      setNetwork={setNetwork}
+      readonly={!isEditing}
+    />
+  ) : tab === 'rpc-url-reset' ? (
+    <RpcUrlResetConfirmation
+      onBack={() => setTab('details')}
+      onSubmit={resetRpcUrlHandler}
+    />
+  ) : tab === 'save' ? (
+    <NetworkUpdateConfirmation
+      onBack={() => setTab('details')}
+      onSubmit={saveHandler}
+    />
+  ) : tab === 'delete' ? (
+    <NetworkDeleteConfirmation
+      onBack={() => setTab('details')}
+      onSubmit={deleteHandler}
+    />
+  ) : null;
```

Comment:
I think the ternary operator makes it harder to read. I'd go either for a `switch` or `if (...) else if (...) else (...)`:

```tsx

if (!network) {
    return <NetworkNotFound />;
}

switch (tab) {
  case 'details':
    return ...;
  case 'rpc-headers':
    return ...;
  // etc...
}
```